static/src/javascripts/bootstraps/enhanced/preferences.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+import React from 'react/addons';
+import {
+    getPopularFiltered,
+    deleteFromSummary,
+    showInMegaNav,
+    showInMegaNavEnable,
+    showInMegaNavEnabled,
+} from 'common/modules/onward/history';
+import isEmpty from 'lodash/objects/isEmpty';
+
+const init = (): void => {
+    const placeholder: ?HTMLElement = document.getElementById(
+        'preferences-history-tags'
+    );
+
+    const initialiseSummaryTagsSettings = () => {
+        const SummaryTagsList: React = React.createClass({
+            getInitialState() {
+                return {
+                    popular: getPopularFiltered(),
+                };
+            },
+            handleRemove(tag: string) {
+                deleteFromSummary(tag);
+                this.setState({
+                    popular: getPopularFiltered({ flush: true }),
+                });
+                showInMegaNav();
+            },
+            render() {
+                const self = this;
+
+                const tags: Object = this.state.popular.reduce((obj, tag) => {
+                    obj[tag[0]] = React.DOM.span(
+                        {
+                            className:
+                                'button button--small button--tag button--secondary',
+                        },
+                        React.DOM.button(
+                            {
+                                onClick: self.handleRemove.bind(self, tag[0]),
```

Comment:
I think we can now call `this.handleRemove.bind(this, tag[0])`, since the reducer is an arrow function. Maybe! 😅